crates/sui-json-rpc/src/governance_api.rs

```diff
@@ -34,39 +37,124 @@ impl GovernanceReadApi {
+                    .or_default()
+                    .push(s);
+                pools
+            });
+
+        let system_state: SuiSystemStateSummary =
+            self.get_system_state()?.into_sui_system_state_summary();
+        let mut delegated_stakes = vec![];
+        for ((pool_id, validator_address), stakes) in pools {
+            let rate_table = self
+                .get_exchange_rate_table(&system_state, &pool_id)
+                .await?;
+
+            let current_rate = self
+                .get_exchange_rate(rate_table, system_state.epoch)
+                .await?;
+
+            let mut delegations = vec![];
+            for stake in stakes {
+                // delegation will be active in next epoch
+                let status = if system_state.epoch >= stake.request_epoch() {
+                    let stake_rate = self
+                        .get_exchange_rate(rate_table, stake.request_epoch())
+                        .await?;
+                    let estimated_reward = (((stake_rate.rate() / current_rate.rate()) - 1.0)
+                        * stake.principal() as f64)
+                        .round() as u64;
+                    StakeStatus::Active { estimated_reward }
+                } else {
+                    StakeStatus::Pending
+                };
+                delegations.push(Stake {
+                    staked_sui_id: stake.id(),
+                    stake_request_epoch: stake.request_epoch(),
+                    // TODO: this might change when we implement warm up period.
+                    stake_active_epoch: stake.request_epoch() + 1,
+                    principal: stake.principal(),
+                    token_lock: stake.sui_token_lock(),
+                    status,
+                })
+            }
+
+            delegated_stakes.push(DelegatedStake {
+                validator_address,
+                staking_pool: pool_id,
+                stakes: delegations,
+            })
+        }
+        Ok(delegated_stakes)
+    }
+
+    fn get_system_state(&self) -> Result<SuiSystemState, Error> {
+        Ok(self.state.database.get_sui_system_state_object()?)
+    }
+
+    async fn get_exchange_rate_table(
+        &self,
+        system_state: &SuiSystemStateSummary,
+        pool_id: &ObjectID,
+    ) -> Result<ObjectID, Error> {
+        let active_rate = system_state.active_validators.iter().find_map(|v| {
+            if &v.staking_pool_id == pool_id {
+                Some(v.exchange_rates_id)
+            } else {
+                None
+            }
+        });
+
+        if let Some(active_rate) = active_rate {
+            return Ok(active_rate);
+        } else {
+            // try find from inactive pool
+            let inactive_validators = system_state.inactive_pools_id;
+            let inactive_validators = self
+                .state
+                .read_table_value::<ObjectID, ValidatorV1>(inactive_validators, pool_id)
```

Comment:
This is not going to work once we have a new version for the Validator type.
There are two ways to go about this:
1. Rely on SuiSystemStateSummary exclusively.
2. Add a trait function to SuiSystemStateTrait to get some basic info to construct this, but then be mindful that that API will have to be respected for all future versions